src/main/resources/data/minecolonies/tags/items/stonemason_product.json

```diff
@@ -8,6 +8,7 @@
     "#minecraft:stairs",
     "#minecraft:walls",
     "minecraft:bricks",
+    "minecraft:furnace",
```

Comment:
So, I get resolving the lack of Furnace. What I don't get is why #forge:stone is a product, when that is a product of the stonesmeltery.  Cobblestone too, as there isn't a normal recipe for the stonemason to produce cobble. 

I expected both #forge:stone and #forge:cobblestone to move to stonemason_ingredient.json. At which point, many of these other items are no longer necessary. 

You also likely want to add redstone to the stonemason_ingredient_excluded.json

Reply:
Using [stone.json](src/generated/resources/data/forge/tags/items/stone.json) as a reference you will see that cobblestone is not included but polished stones are. Also some mods make granite, diorite, and andesite craftable so this could include those. Also the stone smeltery can only handle furnace recipes so the mason handles all stone crafting recipes.

Reply:
I'm not sure which file that hyperlink is supposed to be, as it appears to be a broken link.  My point above is that according to the tags, cobblestone/stone are products of the stonemason, instead of ingredients. Given that they aren't marked as ingredients, you have to add furnace to the products to be able to enable it. 

My proposal is above: add 2 things to stonemason_ingredient.json and 1 to stonemason_ingredient.excluded.json. I believe that fits the theming of the hut, and provides for a much wider array of recipes. I was thinking it might be good to also add #stonemason_product to stonemasion_ingredient.json, which should cut down on future maintenance here.

If you take my proposal, I believe there are several other things in stonemason_product.json that can be removed. 

As an aside, is there a way to provide custom versions of these files for packs? Whatever rules are in them can be broken by 'expert mode' packs

Reply:
Oh! please consider moving minecraft:nether_brick to stonemason_ingredients, that would unblock nether_brick_fence.

Reply:
If you move cobblestone/stone to the ingredients just look at all of the exceptions you would have to add: https://minecraft.gamepedia.com/Cobblestone and https://minecraft.gamepedia.com/Stone.

Yes, the whole reason that we migrated to the use of item tags is because any pack owner or user can easily adjust them with json files and no code updates required.

Reply:
I'm not seeing that many exceptions. I see an ingredient exception for redstone, and one I missed which is stick. 

That leaves comparator and stonecutter unaccounted for, which I'd argue that anywhere you have a redstone exception, a redstone torch should be there.  It might be nice to account for that by including #mechanic_ingredient as stonecutter ingredient exceptions. That does mean that tag marking has to make at least a second pass over things it's already marked, or some form of recursive evaluation.